### PR TITLE
Pull Request for Issue2171: PGM Scanning - XAS

### PR DESCRIPTION
**Pull request by dretrex**
Implemented XAS scanning.  I created the scan configuration, controller, and a shell of a view.  I had to make a PGMPicoAmmeter which uses the synchronized dwell time.

The scan configuration view still needs to be done though as well as some finer polish.

### CODE REVIEW

File: source/application/PGM/PGMAppController.cpp [hunk 1]
```diff
@@ -84,12 +88,27 @@ void PGMAppController::registerExporterOptions()
 
 void PGMAppController::setupScanConfigurations()
 {
-
+	xasScanConfiguration_ = new PGMXASScanConfiguration;
+	xasScanConfiguration_->setControl(0, AMBeamline::bl()->exposedControlByName("Energy")->toInfo());
+//	xasScanConfiguration_->addDetector(AMBeamline::bl()->exposedDetectorByName("teyBladeCurrentDetector")->toInfo());
```
Comment (iainworkman) [inline source/application/PGM/PGMAppController.cpp]: I assume these are being left in as we're not yet certain which detector's Lucia is going to require for the XAS scans?

File: source/ui/dataman/AMScanAxisView.cpp [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#include "AMScanAxisView.h"
+
+
+AMScanAxisView::AMScanAxisView(const QString &title, AMStepScanConfiguration *configuration, QWidget *parent)
+	: QWidget(parent)
+{
+
+}
```
Comment (iainworkman) [inline source/ui/dataman/AMScanAxisView.cpp]: To be done in #2374

**Comment by iainworkman** [comment]
One comment. Given the trivial nature of the comment, and the fact that everything else looks good I will merge.
Time Estimate: 30 minutes.

File: source/application/PGM/PGMAppController.cpp (hunk 1 above)
Comment (dretrex) [inline source/application/PGM/PGMAppController.cpp]: Yup.

**Pull request merged**